deeplearning4j-scaleout/deeplearning4j-scaleout-parallelwrapper/src/main/java/org/deeplearning4j/parallelism/ParallelWrapper.java

```diff
@@ -62,14 +86,17 @@
     protected StatsStorageRouter storageRouter;
     protected boolean isMQ;
     protected WorkspaceMode workspaceMode;
-    private Object[] trainerContextArgs;
+    protected Object[] trainerContextArgs;
+
+    @Getter @Setter protected GradientsAccumulator gradientsAccumulator;
 
     private MagicQueue mq;
 
     // log uncaught exceptions
     Thread.UncaughtExceptionHandler handler = new Thread.UncaughtExceptionHandler() {
         public void uncaughtException(Thread th, Throwable ex) {
             log.error("Uncaught exception: " + ex);
```

Comment:
It doesn't really matter and I'm being pedantic here, but generally with logging we should use ```log.error("Uncaught exception", ex);```
https://www.slf4j.org/faq.html#paramException